naga/src/back/wgsl/writer.rs

```diff
@@ -363,6 +366,32 @@ impl<W: Write> Writer<W> {
         Ok(())
     }
 
+    /// Writes all the necessary directives out
+    fn write_directives(&mut self, module: &Module) -> BackendResult {
+        // Check for dual source blending.
+        if module.types.iter().any(|(_handle, ty)| {
+            if let TypeInner::Struct { ref members, .. } = ty.inner {
+                members.iter().any(|member| {
+                    member.binding.as_ref().is_some_and(|binding| {
+                        matches!(
+                            binding,
+                            &crate::Binding::Location {
+                                blend_src: Some(_),
+                                ..
+                            }
+                        )
+                    })
+                })
+            } else {
+                false
+            }
+        }) {
+            writeln!(self.out, "enable dual_source_blending;")?;
+        }
+
+        Ok(())
```

Comment:
thought: Maybe it would be good to have a blank line written out if we did write anything? Doesn't matter or block, OFC.

Reply:
Not following why we'd write a blank line that case, what's the rationale?

Reply:
The idea is a visual separation in likely the same way that most folks will author WGSL:

```wgsl
// before
enable dual_source_blending;
@compute // …
fn main() {
  // …
}
```

```wgsl
// after
enable dual_source_blending;

@compute // …
fn main() {
  // …
}
```